src/tests/functional/page_refresh_tests.js

```diff
@@ -204,7 +204,7 @@ test("frames marked with refresh='morph' are excluded from full page morphing",
   await expect(page.locator("#refresh-morph")).toHaveText("Loaded morphed frame")
 })
 
-test("navigated frames without refresh attribute are reset after morphing", async ({ page }) => {
```

Comment:
I'm sorry if I explained wrong. Turbo frames that aren't reloaded by morphing have to be reset correctly. This test should be fine.

Turbo frames that don't have the conditions to be reloaded with morphing are affected in the first round of morphing, therefore, they are morphed according to the changes coming from the server. Turbo frames that can be reloaded with morphing due to the presence of the `src` attribute and `refresh="morph"` are ignored in the first round, and then refreshed using morphing.

Reply:
I was just about to send a long final comment saying I finished. :) Happy to do this either way. This test is a bit confusing so it took me a bit to fully make sense of. In your longer explanation, you said what my expectation would be. Specifically:

> This is because we are interested in preserving the URL where the Turbo frame is located if the user has navigated through it. Otherwise, each page refresh would reset the Turbo frame to the initial state of the page, and that isn't what we want.

This test was enforcing the opposite, so I flipped it to reflect what you said here. ^ And it fits my intuition that if a user navigations a turbo frame, and it's implicitly reload="refresh" (not morph) then I would expect that navigation to stay. But you think we should revert it back since it **started** with no src when the page loaded?

Reply:
Yes, that's correct, but only for Turbo frames marked with `refresh="morph"` attribute. Sorry if I wasn't clear enough.

Reply:
What I mean is that the goal of the test as it was before was correct. Turbo frames that don't contain the `refresh="morph"` attribute are reset. We only retain the state of Turbo frames that are reloaded with morphing. The rest are morphed in the first round according to what the server returns in the request made at the time of the page refresh.

Reply:
The test has been reverted (so that it fails again) and the logic has been corrected (so now the test passes).

I think that's it and things are now ready. 🤞